azure-log-analytics/microsoft_api_auth.py

```diff
@@ -28,236 +29,106 @@
 
 # grant types
 CLIENT_CREDENTIALS = 'client_credentials'
-AUTHORIZATION_CODE = 'authorization_code'
-REFRESH_TOKEN = 'refresh_token'
 
 
 class MicrosoftAuth:
-
     def __init__(self, config):
         self.client_id = config.get("client_id")
         self.client_secret = config.get("client_secret")
         self.verify_ssl = config.get('verify_ssl')
-        self.host = config.get("resource")
+        self.scope = config.get('scope', MANAGE_SCOPE)
+        self.host = "https://"
         if self.host[:7] == "http://":
```

Comment:
If host is always use https then we don't need to check again that host is start from http or https.